packages/babel-core/src/transformation/file/index.js

```diff
@@ -54,6 +54,7 @@ export default class File extends Store {
       sourceType:    this.opts.sourceType,
       filename:      this.opts.filename,
       sourceFileName:this.opts.filename,
+      allowReturnOutsideFunction: true,
```

Comment:
This is temporary until https://github.com/babel/babel/pull/3561 is ready and merged